casper/src/main/scala/coop/rchain/casper/LastFinalizedBlockCalculator.scala

```diff
@@ -8,14 +8,12 @@ import coop.rchain.casper.util.ProtoUtil
 import coop.rchain.shared.{Cell, Log}
 import coop.rchain.catscontrib.ListContrib
 
-object LastFinalizedBlockCalculator {
-  // TODO: Extract hardcoded fault tolerance threshold
-  private val faultToleranceThreshold = 0f
-
-  def run[F[_]: Monad: Log: Sync: Concurrent: BlockStore: BlockDagStorage: SafetyOracle](
-      dag: BlockDagRepresentation[F],
-      lastFinalizedBlockHash: BlockHash
-  )(implicit state: Cell[F, CasperState]): F[BlockHash] =
+final case class LastFinalizedBlockCalculator[F[_]: Sync: Log: Concurrent: BlockStore: BlockDagStorage: SafetyOracle](
```

Comment:
Why is this a `case class`? Please make it a regular class. You can add a smart constructor in the companion `object` if your motivation was a nicer syntax.